### PR TITLE
Gradle build in GitHub actions on push to master; spotless code formatting plugin; DOI badge in readme

### PR DESCRIPTION
**Pull request by JonasSchaub**
For this PR, I have added a DOI badge in the readme, added a .yml file to .github/workflows to run a Gradle build in GitHub actions whenever there is a push or PR on master, and added the spotless Gradle plugin for automatic code formatting.

I tried to configure spotless so that the changes to our current formatting are minimal.

### CODE REVIEW

**Comment by JonasSchaub** [comment]
@FelixBaensch we can discuss this tomorrow

File: src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/IMoleculeFragmenter.java [hunk 1]
```diff
@@ -22,12 +22,14 @@
 
 /**
  * TODO:
- * - 
+ * -
  */
```
Comment (FelixBaensch) [inline src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/IMoleculeFragmenter.java:26]: Can we remove this TODO?

**Review by FelixBaensch — APPROVED** [review]

File: src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/IMoleculeFragmenter.java (hunk 1 above)
Comment (JonasSchaub) [reply src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/IMoleculeFragmenter.java]: Done in latest commit.

**Review by FelixBaensch — APPROVED** [review]

**Comment by FelixBaensch** [comment]
Thank you Jonas for your excellent work.

**Pull request merged**